_posts/2022-04-21-tardigrade-launch.md

```diff
@@ -0,0 +1,323 @@
+   * `<spool-bucket-name>` - name of a bucket to be used for spooling.
+     Multiple buckets can be specified using comma,
+     e.g.: `s3://bucket-1,s3://bucket-2`
+   * `<aws-region>` - aws region where the cluster is being deployed,
+     e.g.: `us-east-1`
+   * `<permanent-data-bucket>` - name of a bucket to be used for storing tables
+     created by Trino
+   * `<access-key>`, `<secret-key>` - AWS access credentials. The key must have
+     full access to S3 buckets specified for spooling and for permanent storage
+     as well as access to `Glue`
+4. Create a Kubernetes cluster based on the recommended hardware configuration
+   with `eksctl`:
+   ```bash
+    eksctl create cluster \
+      --name <cluster-name> \
+      --region <aws-region> \
+      --node-type m5.8xlarge \
+      --nodes <desired-number-of-workers + 1>
+   ```
+   For example:
+   ```bash
+    eksctl create cluster \
+      --name tardigrade-cluster \
+      --region us-east-1 \
+      --node-type m5.8xlarge \
+      --nodes 6
+   ```
+5. Deploy a Trino cluster:
+   ```bash
+   helm upgrade --install \
+     --values values.yaml \
+     tardigrade-kubernetes-cluster \
+     trino/trino \
+     --version 0.7.0
+
+   kubectl rollout restart deployment \
+     tardigrade-testing-batch-trino-coordinator \
+     tardigrade-testing-batch-trino-worker
+   ```
+6. Forward the Trino endpoint to your local machine:
+   ```bash
+   kubectl port-forward \
+   $(kubectl get pods --namespace default -l "app=trino,release=tardigrade-testing-batch,component=coordinator" -o jsonpath="{.items[0].metadata.name}") \
+   8080:8080
+   ```
+7. Run some queries:
+   ```bash
+   ./trino-cli-*-executable.jar --server 127.0.0.1:8080
+   ```
+   Trino CLI can be downloaded from [here](https://trino.io/download.html)
+8. Terminate Kubernetes cluster:
+   ```bash
+   eksctl delete cluster \
+     --name <cluster-name> \
+     --region <aws-region>
+   ```
+   For example:
+   ```bash
+   eksctl delete cluster \
+     --name tardigrade-cluster \
+     --region us-east-1
+   ```
+
+### Potential issues
+
+* Queries may start failing with
+  `software.amazon.awssdk.services.s3.model.S3Exception: Please reduce your request rate`:
+  * If your workload is I/O intensive it is possible that the S3 may start
+    throttling some of the requests. The request limits are enforced per
+    bucket. It is recommended to create an another bucket for spooling to allow
+    Trino to balance the load
+    `exchange.base-directories=s3://<first-bucket>,s3://<second-bucket>`
+* Queries may fail with `Task descriptor storage capacity has been exceeded`:
+  * Trino has to maintain descriptors for each task in case it has to be
+    restarted in an event of a failure. This information is currently stored
+    in memory on coordinator. In the future we are planning to implement
+    adaptive spilling for task descriptors, but at the moment there's a
+    chance queries may hit this limit. If this happens we recommend:
```

Comment:
we recommend that you run less queries concurrently or use a coordinator with more memory.